src/common/dataHooks/useSearchLoader.js

```diff
@@ -0,0 +1,117 @@
+import { useRef } from 'react'
+import { useInfiniteQuery } from 'react-query'
+import { useParams } from 'react-router-dom'
+import PropTypes from 'prop-types'
+
+// FPCC
+import useIntersectionObserver from 'common/hooks/useIntersectionObserver'
+import api from 'services/api'
+import {
+  SEARCH,
+  TYPE_PHRASE,
+  TYPE_SONG,
+  TYPE_STORY,
+  TYPE_WORD,
+} from 'common/constants'
+
+/**
+ * Calls search API and provides search results and infinite scroll info.
+ */
+function useSearchLoader({ searchParams }) {
+  const { sitename } = useParams()
+  const searchParamString = searchParams.toString()
+
+  const pagesDataAdaptor = (pages) =>
+    pages.map((page, index) => singlePageDataAdaptor(page, index))
+
+  const singlePageDataAdaptor = (page, index) => {
+    const formattedEntries = page?.results?.map((result) =>
+      resultAdaptor(result),
+    )
+    return {
+      ...page,
+      pageNumber: index + 1,
+      results: formattedEntries,
+    }
+  }
+
+  const resultAdaptor = (result) => {
+    const baseObject = {
+      id: result?.entry?.id,
+      title: result?.entry?.title,
+      type: result?.type,
+      sitename: result?.entry?.site?.slug,
+      siteTitle: result?.entry?.site?.title,
+      visibility: result?.entry?.visibility,
+    }
+    switch (result?.type) {
+      case 'dictionary_entry':
+      case TYPE_WORD:
+      case TYPE_PHRASE:
+        return {
+          ...baseObject,
+          translations: result?.entry?.translations,
+          audio: result?.entry?.relatedAudio,
```

Comment:
The other properties are not included because they aren't used in search result display, right?